spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraJoinRDD.scala

```diff
@@ -34,7 +34,9 @@ class CassandraJoinRDD[L, R] private[connector](
     val where: CqlWhereClause = CqlWhereClause.empty,
     val limit: Option[Long] = None,
     val clusteringOrder: Option[ClusteringOrder] = None,
-    val readConf: ReadConf = ReadConf())(
+    val readConf: ReadConf = ReadConf(),
+    manualRowReader: Option[RowReader[R]] = None,
+    manualRowWriter: Option[RowWriter[L]] = None)(
```

Comment:
This constructor is private, so it is not exposed directly, so have you considered:
 - removing implicit RowReaderFactory / RowWriterFactory from here at all,
 - making rowReader / rowWriter obligatory
 - evaluating rowReader / rowWriter at the call site, right before newing this class?